src/data/finset/lattice.lean

```diff
@@ -111,6 +111,20 @@ begin
   exact ⟨λ h c hc b hb, h b hb c hc, λ h b hb c hc, h c hc b hb⟩,
 end
 
+lemma le_sup_of_mem [decidable_eq α] {a : α} (as : a ∈ s.image f) :
+  a ≤ s.sup f :=
+begin
+  rcases mem_image.mp as with ⟨w, ws, rfl⟩,
+  exact le_sup ws,
+end
+
+lemma le_sup_of_mem_set {a : α} (as : a ∈ f '' s) :
```

Comment:
At this point I'm not at all convinced that these two sup lemmas are useful; I assume they're now unused?

Reply:
Ok, they are unused and I removed them.